### PR TITLE
docs: add documentation for notifications

### PR DESCRIPTION
**Pull request by jaideepr97**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
> /kind enhancement
 /kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:
updates documentation for argocd-notifications

**NOTE** 
This PR should be merged after #638 has been merged

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: docs/reference/api.html.md [hunk 1]
```diff
@@ -2152,6 +2241,22 @@ Unknown: For some reason the state of the Argo CD Dex component could not be obt
 </tr>
 <tr>
 <td>
+<code>notifications</code></br>
```
Comment (ishitasequeira) [inline docs/reference/api.html.md:2244]: image

should we add more space for the column to fit notifications in 1 line 🤔 ?

File: docs/reference/api.html.md [hunk 2]
```diff
@@ -1192,6 +1205,82 @@ ingress supports SNI.</p>
...
+</td>
+<td>
+<p>Replicas determins the number of replicas for notifications controller.</p>
+</td>
+</tr>
+<tr>
+<td>
+<code>resources</code></br>
+<em>
+<a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#resourcerequirements-v1-core">
```
[the remark this answers is not in the document]
Comment (jaideepr97) [reply docs/reference/api.html.md:1262]: fixed

File: docs/reference/api.html.md [hunk 3]
```diff
@@ -2152,6 +2241,22 @@ Unknown: For some reason the state of the Argo CD Dex component could not be obt
...
+string
+</em>
+</td>
+<td>
+<p>Notifications is a simple, high-level summary of where the Argo CD Notifications controller component is in its lifecycle.
+There are five possible notifications values:
+Pending: The Argo CD Notifications controller component has been accepted by the Kubernetes system, but one or more of the required resources have not been created.
+Running: All of the required Pods for the Argo CD Notifications controller component are in a Ready state.
+Failed: At least one of the  Argo CD Notifications controller component Pods had a failure.
+Unknown: For some reason the state of the Argo CD Notifications controller component could not be obtained.</p>
```
[the remark this answers is not in the document]
Comment (jaideepr97) [reply docs/reference/api.html.md:2255]: fixed

File: docs/reference/api.html.md (hunk 1 above)
Comment (jaideepr97) [reply docs/reference/api.html.md:2244]: done, thanks @reginapizza !

**Review by ishitasequeira — APPROVED** [review]
LGTM!

**Review by jopit — APPROVED** [review]
LGTM

**Pull request merged**